contracts/libs/Tx.sol

```diff
@@ -0,0 +1,516 @@
+pragma solidity ^0.5.15;
+pragma experimental ABIEncoderV2;
+
+import {BLS} from "./BLS.sol";
+
+library Tx {
+    uint256 public constant MASK_ACCOUNT_ID = 0xffffffff;
+    uint256 public constant MASK_STATE_ID = 0xffffffff;
+    uint256 public constant MASK_AMOUNT = 0xffffffff;
+    uint256 public constant MASK_NONCE = 0xffffffff;
+    uint256 public constant MASK_TOKEN_ID = 0xffff;
+    uint256 public constant MASK_BYTE = 0xff;
+
+    uint256 constant bound4Bytes = 0x10000000000000000;
+    uint256 constant bound2Bytes = 0x100000000;
+    uint256 constant bound1Bytes = 0x10000;
+
+    // transaction_type: transfer
+    // [sender_state_id<4>|receiver_state_id<4>|amount<4>|nonce<4>]
+    uint256 public constant TX_LEN_0 = 16;
+    uint256 public constant MASK_TX_0 = 0xffffffffffffffffffffffffffffffff;
+    // positions in bytes
+    uint256 public constant POSITION_SENDER_0 = 4;
+    uint256 public constant POSITION_RECEIVER_0 = 8;
+    uint256 public constant POSITION_AMOUNT_0 = 12;
+    uint256 public constant POSITION_NONCE_0 = 16;
+
+    // transaction_type: create
+    // [receiver_account_id<4>|receiver_state_id<4>|token<2>]
+    uint256 public constant TX_LEN_1 = 10;
+    uint256 public constant MASK_TX_1 = 0xffffffffffffffffffff;
+    // positions in bytes
+    uint256 public constant POSITION_ACCOUNT_1 = 4;
+    uint256 public constant POSITION_STATE_1 = 8;
+    uint256 public constant POSITION_TOKEN_1 = 10;
+
+    // transaction_type: burn concent
+    // [burner_state_id<4>|amount<4>|nonce<4>|sign<1>]
+    uint256 public constant TX_LEN_2 = 13;
+    uint256 public constant MASK_TX_2 = 0xffffffffffffffffffffffffff;
+    // positions in bytes
+    uint256 public constant POSITION_STATE_2 = 4;
+    uint256 public constant POSITION_AMOUNT_2 = 8;
+    uint256 public constant POSITION_NONCE_2 = 12;
+    uint256 public constant POSITION_SIGN_2 = 13;
+
+    // transaction_type: airdrop
+    // [receiver_state_id<4>|amount<4>]
+    uint256 public constant TX_LEN_3 = 8;
+    uint256 public constant MASK_TX_3 = 0xffffffff;
+    // positions in bytes
+    uint256 public constant POSITION_RECEIVER_3 = 4;
+    uint256 public constant POSITION_AMOUNT_3 = 8;
+
+    struct TransferDecoded {
+        uint256 senderID;
+        uint256 receiverID;
+        uint256 amount;
+        uint256 nonce;
+    }
```

Comment:
It looks like our transaction has 3 types of struct: the encoded, the encoded with sig, and the compressed.
Is the TransferDecoded the encoded with sig one that is sent to the fraudproof? Do we need all fields for this? i.e.

```
    struct Transaction {
        uint256 fromIndex;
        uint256 toIndex;
        uint256 tokenType;
        uint256 nonce;
        uint256 txType;
        uint256 amount;
        bytes signature;
    }
```
@vaibhavchellani What's your thoughts on this?